kubectl-cloudflow/cfapp/cloudflowapplication.go

```diff
@@ -299,11 +312,38 @@ func (in *CloudflowApplication) GenerateOwnerReference() metav1.OwnerReference {
 	managingController := true
 	blockOwnerDeletion := true
 	return metav1.OwnerReference{
-		APIVersion:         ApiVersion(),
+		APIVersion:         APIVersion(),
 		Kind:               Kind,
 		Name:               in.GetName(),
 		UID:                in.GetUID(),
 		Controller:         &managingController,
 		BlockOwnerDeletion: &blockOwnerDeletion}
 
 }
+
+//CheckApplicationDescriptorVersion checks the version, logs and exits if not supported
+func CheckApplicationDescriptorVersion(spec CloudflowApplicationSpec) {
+	if spec.Version != SupportedApplicationDescriptorVersion {
+		// If the version is an int, compare them, otherwise provide a more general message.
+		if version, err := strconv.Atoi(spec.Version); err == nil {
+			if supportedVersion, err := strconv.Atoi(SupportedApplicationDescriptorVersion); err == nil {
+				if version < supportedVersion {
+					if spec.LibraryVersion != "" {
+						printutil.LogAndExit("Application built with sbt-cloudflow version '%s', is incompatible and no longer supported. Please upgrade sbt-cloudflow and rebuild the application with 'sbt buildApp'.", spec.LibraryVersion)
+					} else {
+						printutil.LogAndExit("Application is incompatible and no longer supported. Please upgrade sbt-cloudflow and rebuild the application with 'sbt buildApp'.")
+					}
+				}
+				if version > supportedVersion {
+					if spec.LibraryVersion != "" {
+						printutil.LogAndExit("Application built with sbt-cloudflow version '%s', is incompatible and requires a newer version of the kubectl cloudflow plugin. Please upgrade and try again.", spec.LibraryVersion)
+					} else {
+						printutil.LogAndExit("Application is incompatible and requires a newer version of the kubectl cloudflow plugin. Please upgrade and try again.")
+					}
+				}
+			}
+		}
+
+		printutil.LogAndExit("Application is incompatible and no longer supported. Please update sbt-cloudflow and rebuild the application with 'sbt buildApp'.")
+	}
+}
```

Comment:
should we exit from this function if the check fails ? Or have this function return an error message and check for `nil` in the caller ?

Reply:
I'll change it to error cause it is used somewhere else and can probably be made private now.

Reply:
I've changed it now so that all LogAndExit and LogErrorAndExit calls are only done in commands.